src/Illuminate/Database/Query/Builder.php

```diff
@@ -31,6 +31,15 @@
 
 use function Illuminate\Support\enum_value;
 
+/**
+ * @method self where(string $column, $value)
+ * @method self where(string $column, string $operator, $value)
+ * @method self where(string $column, string $operator, $value, 'and'|'or' $boolean)
+ * @method self where(\Closure(): mixed)
+ * @method self where(array<string, mixed> $column)
+ * @method self where((array{string, 'and'|'or', mixed}|array{string, mixed}|array{column: string, operator?: 'and'|'or', value: mixed})[] $column)
+ * @method self where(\Closure|string|array|\Illuminate\Contracts\Database\Query\Expression $column, mixed $operator  null, mixed $value = null, string $boolean = 'and')
+ */
```

Comment:
Last time I checked, which was around 6 months ago, a lot of tooling did not support literals / generics in `@method` docblocks.

```suggestion
/**
 * @method self where(string $column, $value)
 * @method self where(string $column, string $operator, $value)
 * @method self where(string $column, string $operator, $value, string $boolean)
 * @method self where(\Closure(): mixed)
 * @method self where(array $column)
 * @method self where(\Closure|string|array|\Illuminate\Contracts\Database\Query\Expression $column, mixed $operator = null, mixed $value = null, string $boolean = 'and')
 */
```